li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/producer/LiKafkaProducerImpl.java

```diff
@@ -301,6 +304,18 @@ private LiKafkaProducerImpl(LiKafkaProducerConfig configs,
         }
       }
 
+      // TODO: discuss with navina if adding enableRecordHeader in header to pass parameter is a good choice.
+      // It may be a little strange to control if using record headers by testing if the record header has such a field.
+      if (_enableRecordHeader) {
```

Comment:
This is a producer instance-level configuration and hence, doesn't need to be inside every record send. 
You can check for the config in the producer constructor and pass the value as a boolean argument to the `MessageSplitterImpl` constructor.

Reply:
Yeah, it should be a producer instance-level configuration.  I will update to put the config in the constructor.